# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/prepare.rs
# Commit: 5ccaf8100f79
# Full Hash: 5ccaf8100f7912e55bb98cc1951d57df9c4dd511
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2022-04-26 16:04:35
# Regressor Bug: 1766017
# File Overlap Count: 1
# Description:
#   Bug 1766017 - Do a better job of clipping off-screen surface allocation sizes r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D144534
# ==============================================================================

diff -r 01d5f09a02bf -r 5ccaf8100f79 gfx/wr/webrender/src/prepare.rs
--- a/gfx/wr/webrender/src/prepare.rs	Tue Apr 26 09:24:16 2022 +0000
+++ b/gfx/wr/webrender/src/prepare.rs	Tue Apr 26 09:42:22 2022 +0000
@@ -761,7 +761,7 @@
                     let splitter = &mut frame_state.plane_splitters[plane_splitter_index.0];
                     let surface_index = pic.raster_config.as_ref().unwrap().surface_index;
                     let surface = &frame_state.surfaces[surface_index.0];
-                    let local_prim_rect = surface.local_rect.cast_unit();
+                    let local_prim_rect = surface.clipped_local_rect.cast_unit();
 
                     PicturePrimitive::add_split_plane(
                         splitter,